vllm/outputs.py

```diff
@@ -57,14 +61,21 @@ class PoolingOutput:
     """The output data of one pooling output of a request.
 
     Args:
-        embedding: The embedding vector, which is a list of floats. The
-        length of vector depends on the model as listed in the embedding guide.
+        data: The extracted hidden states.
     """
-    embedding: List[float]
+    data: torch.Tensor
```

Comment:
We now return a tensor directly. It is up to the task-specific output classes (e.g. `EmbeddingOutput`) to parse the tensor into a format that is convenient to the user.

Reply:
Would it make sense to return a list of tensors here, in case the output of the model is multiple tensors? 
Or do we assume that in such case the model works stack any output tensors into one before returning them?

Reply:
The type annotation here can be changed without affecting the internals (originally I wanted to be even more abstract and annotate this as type `object`). For now, all models return a single tensor, so that's what I put there.